core/domain/user_query_domain.py

```diff
@@ -19,26 +19,43 @@
 from __future__ import annotations
 
 import collections
+import datetime
 
 from core import feconf
 from core import utils
 from core.constants import constants
 
-attribute_names = [ # pylint: disable=invalid-name
-    predicate['backend_attr'] for predicate in (
-        constants.EMAIL_DASHBOARD_PREDICATE_DEFINITION)]
+from typing import List, Union
 
-UserQueryParams = collections.namedtuple( # pylint: disable=invalid-name
-    'UserQueryParams', attribute_names, defaults=(None,) * len(attribute_names))
+attribute_names = [ # pylint: disable=invalid-name
+        predicate['backend_attr'] for predicate in (
+            constants.EMAIL_DASHBOARD_PREDICATE_DEFINITION)]
+
+# We can't inject dynamic data(attribute_names) into namedtuple(UserQueryParams)
+# without raising mypy error. Therefore , we need to silence these errors
+# explicitly using # type: ignore[misc] for now.
+# For more information : https://github.com/python/mypy/issues/848
+UserQueryParams = collections.namedtuple( # type: ignore[misc] # pylint: disable=invalid-name
+        'UserQueryParams',
+        attribute_names,
+        defaults=(None,) * len(attribute_names) # type: ignore[misc]
+        )
 
 
 class UserQuery:
     """Domain object for the UserQueryModel."""
 
     def __init__(
-            self, query_id, query_params, submitter_id, query_status, user_ids,
-            sent_email_model_id=None, created_on=None, deleted=False
-    ):
+            self,
+            query_id: str,
+            query_params: UserQueryParams,
+            submitter_id: str,
+            query_status: str,
+            user_ids: List[str],
+            sent_email_model_id: Union[str, None]=None,
+            created_on: Union[datetime.datetime, None]=None,
```

Comment:
Maybe use `Optional`?

Reply:
Sure !!  But it would add an extra unnecessary import ?

Reply:
I think it is fine to add more typing imports.

Reply:
Sure !!